app/src/main/java/de/westnordost/streetcomplete/data/AndroidDatabase.kt

```diff
@@ -3,11 +3,33 @@ package de.westnordost.streetcomplete.data
 import android.annotation.SuppressLint
 import android.content.ContentValues
 import android.database.Cursor
-import android.database.sqlite.SQLiteDatabase.*
+import android.database.sqlite.SQLiteDatabase.CONFLICT_ABORT
+import android.database.sqlite.SQLiteDatabase.CONFLICT_FAIL
+import android.database.sqlite.SQLiteDatabase.CONFLICT_IGNORE
+import android.database.sqlite.SQLiteDatabase.CONFLICT_NONE
+import android.database.sqlite.SQLiteDatabase.CONFLICT_REPLACE
+import android.database.sqlite.SQLiteDatabase.CONFLICT_ROLLBACK
 import android.database.sqlite.SQLiteOpenHelper
 import android.database.sqlite.SQLiteStatement
-import de.westnordost.streetcomplete.data.ConflictAlgorithm.*
-import de.westnordost.streetcomplete.ktx.*
+import de.westnordost.streetcomplete.data.ConflictAlgorithm.ABORT
+import de.westnordost.streetcomplete.data.ConflictAlgorithm.FAIL
+import de.westnordost.streetcomplete.data.ConflictAlgorithm.IGNORE
+import de.westnordost.streetcomplete.data.ConflictAlgorithm.REPLACE
+import de.westnordost.streetcomplete.data.ConflictAlgorithm.ROLLBACK
+import de.westnordost.streetcomplete.ktx.getBlob
+import de.westnordost.streetcomplete.ktx.getBlobOrNull
+import de.westnordost.streetcomplete.ktx.getDouble
+import de.westnordost.streetcomplete.ktx.getDoubleOrNull
+import de.westnordost.streetcomplete.ktx.getFloat
+import de.westnordost.streetcomplete.ktx.getFloatOrNull
+import de.westnordost.streetcomplete.ktx.getInt
+import de.westnordost.streetcomplete.ktx.getIntOrNull
+import de.westnordost.streetcomplete.ktx.getLong
+import de.westnordost.streetcomplete.ktx.getLongOrNull
+import de.westnordost.streetcomplete.ktx.getShort
+import de.westnordost.streetcomplete.ktx.getShortOrNull
+import de.westnordost.streetcomplete.ktx.getString
+import de.westnordost.streetcomplete.ktx.getStringOrNull
```

Comment:
Why are wildcard imports considered to be bad?

Reply:
Wildcard imports make it harder to browse the source code as plain text. Explicit imports means I can search in the file to find where something is imported from.

Why are explicit imports bad? That's a rhetorical question; we've had this conversation before, so I know the answer. The point of the question is to show that "you should use better tooling" works in both directions.

A smart IDE can auto-import and collapse the imports, making them effectively equivalent to wildcards, for people using the IDE. People who opt for simple tooling do not have this option, so we should style for them by default.